lib/internal/fs/promises.js

```diff
@@ -129,6 +130,11 @@ function lazyFsStreams() {
   return fsStreams ??= require('internal/fs/streams');
 }
 
+function handleErrorFromBinding(error) {
```

Comment:
It would be useful to have some comments describing why this is necessary.

```suggestion
// By the time the C++ land creates an error for a promise rejection (likely from a
// libuv callback), there is already no JS frames on the stack. So we need to
// wait until V8 resumes execution back to JS land before we have enough information
// to re-capture the stack trace.
function handleErrorFromBinding(error) {
```